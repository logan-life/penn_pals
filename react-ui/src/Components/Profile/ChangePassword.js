import React, { useState } from 'react';
import changePasswordFunct from './changePasswordFunction';
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";


export default function ChangePassword(){
    const[newPwd, setPassword] = useState("");
    const[newPwdCopy,setCopy] = useState("");
    const[oldPwd,setOldPassword] = useState("");
    async function changePassword(e){
      e.preventDefault();
      if(newPwd!==newPwdCopy){
          window.alert('Password and password copy are different');
      }
      else{
      const passwordData = Object.freeze({
        password: oldPwd,
        newPassword:newPwd
    });
    const res = await changePasswordFunct(passwordData);
    if (res.status===200){
      window.alert('Password was successfully changed');
    }
    else{
      const error = new Error(res.error);
      throw error;
    }
 
    }
}
    return(
      <div id = "changePwd">
      <div className="Profile Person">
      <Form.Group controlID="oldPassword">
            <Form.Label htmlFor="oldPassword">Old Password</Form.Label>
            <Form.Control
              type="password"
              value={oldPwd}
              name="password"
              id="oldPassword"
              placeholder="Old password"
              onChange={(e) => setOldPassword(e.target.value)}
            />
          </Form.Group>
          <Form.Group controlID="newPassword">
            <Form.Label htmlFor="newPassword">New Password</Form.Label>
            <Form.Control
              type="password"
              value={newPwd}
              name="newPassword"
              id="newPassword"
              placeholder="New password"
              onChange={(e) => setPassword(e.target.value)}
            />
          </Form.Group>
            <Form.Group controlID="newPasswordCopy">
            <Form.Label htmlFor="newPasswordCopy">Repeat new password</Form.Label>
            <Form.Control
              type="password"
              value={newPwdCopy}
              name="newPasswordCopy"
              id="newPasswordCopy"
              placeholder="Repeat new password" 
              onChange={(e) => setCopy(e.target.value)}
            />
          </Form.Group>
        <Button
            onClick={changePassword}
            type="button"
            id="changePswd"
            variant="outline-primary"
          >
            Change Password
          </Button>
        </div>
        </div>
    );    
    
}

