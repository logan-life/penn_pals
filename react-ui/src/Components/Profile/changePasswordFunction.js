export default async function changePasswordFunct(passwordData) {
  try {
    const resp = await fetch("api/changePassword", {
      method: "PUT",
      body: JSON.stringify(passwordData),
      credentials: "include",
      headers: {
        "Content-Type": "application/json",
      },
    });
    return resp;
  } catch (error) {
    throw new Error(error);
  }
}
