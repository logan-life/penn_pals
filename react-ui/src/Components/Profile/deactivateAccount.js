export default async function deactivateAccount(passwordData) {
  try {
    const res = await fetch("api/profile", {
      method: "PUT",
      body: JSON.stringify(passwordData),
      credentials: "include",
      headers: {
        "Content-Type": "application/json",
      },
    });
    return res;
  } catch (error) {
    throw new Error(error);
  }
}
