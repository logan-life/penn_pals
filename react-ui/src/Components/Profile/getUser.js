export default async function getUser() {
  try {
    const resp = await fetch("api/profile", {
      method: "GET",
      credentials: "include",
      headers: {
        "Content-Type": "application/json",
      },
    });
    return resp;
  } catch (error) {
    throw new Error(error);
  }
}
