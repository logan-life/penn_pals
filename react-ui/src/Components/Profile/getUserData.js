import axios from "axios";
export const getUserData = async (userN, firstN, lastN, pswd) => {
  const resp = await axios.post("http://localhost:8080/api/users/", {
    username: userN,
    firstname: firstN,
    lastname: lastN,
    password: pswd,
  });
  const data = resp;

  return data;
};
