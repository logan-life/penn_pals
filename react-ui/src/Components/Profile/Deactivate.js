import React, { useState } from 'react';
import deactivateAccount from './deactivateAccount';
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";

export default function Deactivate(){
    const[oldPwd,setOldPassword] = useState("");
    async function deactivate(e){
      e.preventDefault();
      const passwordData = Object.freeze({
        password: oldPwd
    });

    const res = await deactivateAccount(passwordData);
    if(res.status===200){
      window.alert('Account was succesfully deactivated');
    }
    else{
      const error = new Error(res.error);
      throw error;
    }
}   
    return(
      <div id = "changePwd">
      <Form.Group controlID="oldPassword">
            <Form.Label htmlFor="oldPassword">Old Password</Form.Label>
            <Form.Control
              type="password"
              value={oldPwd}
              name="password"
              id="password"
              placeholder="Old password"
              onChange={(e) => setOldPassword(e.target.value)}
            />
          </Form.Group>
          <Button
            onClick={deactivate}
            type="button"
            id="deactivate"
            variant="outline-primary"
          >
            Deactivate Account
          </Button>
        </div>
    );    
    
}
