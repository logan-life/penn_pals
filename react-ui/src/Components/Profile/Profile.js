import React, { useEffect, useState } from "react";
import Alert from "react-bootstrap/Alert";
import { BrowserRouter as Router, Link, Route, Switch } from "react-router-dom";
import ChangePassword from "./ChangePassword";
import Deactivate from "./Deactivate";
import ResetPassword from "../Login/ForgotPassword";
import getUser from "./getUser";

export default function Profile() {
  const [userName, setUser] = useState("");
  const [joinDate, setJoinDate] = useState("");
  const [picture, setPicture] = useState(
    "https://s.gravatar.com/avatar/5811b64ffd2fa7116580dafa02d9bb7b?s=200&r=pg&d=mp"
  );
  const [email, setEmail] = useState("");
  const [fName, setFName] = useState("");
  const [lName, setLName] = useState("");
  const [isLoggedIn, setLogIn] = useState(false);

  async function getUserData() {
    const res = await getUser();
    if (res.status === 200) {
      setLogIn(true);
      res.json().then(function (json) {
        setUser(json.username);setPicture(json.avatar);setEmail(json.email);
        setFName(json.firstname);setLName(json.lastname);
        let date = new Date(json.register_time);
        setJoinDate(date.toLocaleDateString());
      });
    }
  }

  useEffect(() => {
    getUserData();
  }, []);

  return isLoggedIn ? (
    <div id="userProf">
      <img
        id="pic"
        src={picture}
        alt="User Avatar"
        width="150"
        height="150"
      ></img>
      <div className="Profile Person">
        <p id="name">
          {fName} {lName}
        </p>
        <p id="useName">{userName}</p>
        <p id="date">{joinDate}</p>
        <p id="email">{email}</p>
      </div>
      <div id="personalView">
        <Router>
          <Link id="changepwd" to="/ChangePassword">
            Change Password
          </Link>
          <Switch>
            <Route path="/ChangePassword">
              <ChangePassword />
            </Route>
          </Switch>
        </Router>
        <br></br>
        <Router>
          <Link id="deactivate" to="/Deactivate">
            Deactivate Account
          </Link>
          <Switch>
            <Route path="/Deactivate">
              <Deactivate />
            </Route>
          </Switch>
        </Router>
        <br></br>
        {/* <Router>
                <Link id="reset" to="/ResetPassword">Reset Password</Link>
                <Switch>
                <Route path='/ResetPassword'>
                  <ResetPassword />
                  </Route>
                </Switch>
                </Router> */}
      </div>
    </div>
  ) : (
    <Alert id="alertContactViewNotLoggedIn" variant="danger">
      Please{" "}
      <Alert.Link as={Link} to="/Login">
        login
      </Alert.Link>{" "}
      to see your profile.
    </Alert>
  );
}
