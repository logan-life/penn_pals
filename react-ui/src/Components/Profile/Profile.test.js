import Profile from './Profile';
import Deactivate from './Deactivate';
import ChangePassword from './ChangePassword';
import {BrowserRouter} from "react-router-dom";
const renderer = require('react-test-renderer');

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

// snapshot testing
test('Profile view matches snapshot', () => {
const component = renderer.create(<BrowserRouter><Profile /></BrowserRouter>);
let tree = component.toJSON();
expect(tree).toMatchSnapshot();
});

// snapshot testing
test('Deactivate view matches snapshot', () => {
const component = renderer.create(<BrowserRouter><Deactivate /></BrowserRouter>);
let tree = component.toJSON();
expect(tree).toMatchSnapshot();
});

// snapshot testing
test('ChangePassword view matches snapshot', () => {
const component = renderer.create(<BrowserRouter><ChangePassword /></BrowserRouter>);
let tree = component.toJSON();
expect(tree).toMatchSnapshot();
});