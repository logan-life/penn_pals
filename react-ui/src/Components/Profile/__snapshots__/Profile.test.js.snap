// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword view matches snapshot 1`] = `
<div
  id="changePwd"
>
  <div
    className="Profile Person"
  >
    <div
      className="form-group"
      controlID="oldPassword"
    >
      <label
        className="form-label"
        htmlFor="oldPassword"
      >
        Old Password
      </label>
      <input
        className="form-control"
        id="oldPassword"
        name="password"
        onChange={[Function]}
        placeholder="Old password"
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
      controlID="newPassword"
    >
      <label
        className="form-label"
        htmlFor="newPassword"
      >
        New Password
      </label>
      <input
        className="form-control"
        id="newPassword"
        name="newPassword"
        onChange={[Function]}
        placeholder="New password"
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
      controlID="newPasswordCopy"
    >
      <label
        className="form-label"
        htmlFor="newPasswordCopy"
      >
        Repeat new password
      </label>
      <input
        className="form-control"
        id="newPasswordCopy"
        name="newPasswordCopy"
        onChange={[Function]}
        placeholder="Repeat new password"
        type="password"
        value=""
      />
    </div>
    <button
      className="btn btn-outline-primary"
      disabled={false}
      id="changePswd"
      onClick={[Function]}
      type="button"
    >
      Change Password
    </button>
  </div>
</div>
`;

exports[`Deactivate view matches snapshot 1`] = `
<div
  id="changePwd"
>
  <div
    className="form-group"
    controlID="oldPassword"
  >
    <label
      className="form-label"
      htmlFor="oldPassword"
    >
      Old Password
    </label>
    <input
      className="form-control"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Old password"
      type="password"
      value=""
    />
  </div>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    id="deactivate"
    onClick={[Function]}
    type="button"
  >
    Deactivate Account
  </button>
</div>
`;

exports[`Profile view matches snapshot 1`] = `
<div
  className="fade alert alert-danger show"
  id="alertContactViewNotLoggedIn"
  role="alert"
>
  Please
   
  <a
    className="alert-link"
    href="/Login"
    onClick={[Function]}
  >
    login
  </a>
   
  to see your profile.
</div>
`;
